### PR TITLE
fix(sql): fix exception updating symbol column with char value

### LINKED ISSUE #3563: Error updating symbol column to single char
### Describe the bug

Update of symbol column to function which return char does not work

```
UPDATE X 
SET sym = NULLIF(CONCAT(to_str(a, 'yyyy-MM-dd'), 'n'), 'n')
```

```
java.lang.UnsupportedOperationException
	at io.questdb.griffin.engine.functions.CharFunction.getSymbol(CharFunction.java:182)
	at io.questdb.cairo.sql.VirtualRecord.getSym(VirtualRecord.java:193)
	at io.questdb.griffin.UpdateOperatorImpl.appendRowUpdate(UpdateOperatorImpl.java:398)
	at io.questdb.griffin.UpdateOperatorImpl.executeUpdate(UpdateOperatorImpl.java:187)
	at io.questdb.griffin.engine.ops.UpdateOperation.apply(UpdateOperation.java:74)
	at io.questdb.cairo.TableWriter.apply(TableWriter.java:652)
	at io.questdb.griffin.CompiledQueryImpl$1.apply(CompiledQueryImpl.java:62)
	at io.questdb.griffin.CompiledQueryImpl$1.apply(CompiledQueryImpl.java:59)
	at io.questdb.griffin.engine.ops.OperationDispatcher.execute(OperationDispatcher.java:67)
	at io.questdb.griffin.CompiledQueryImpl.execute(CompiledQueryImpl.java:86)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.executeUpdate(JsonQueryProcessor.java:574)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.compileQuery(JsonQueryProcessor.java:427)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.execute0(JsonQueryProcessor.java:186)
	at io.questdb.cutlass.http.processors.JsonQueryProcessor.onRequestComplete(JsonQueryProcessor.java:236)
	at io.questdb.cutlass.http.HttpConnectionContext.handleClientRecv(HttpConnectionContext.java:604)
	at io.questdb.cutlass.http.HttpConnectionContext.handleClientOperation(HttpConnectionContext.java:203)
	at io.questdb.cutlass.http.HttpServer$1.lambda$$0(HttpServer.java:86)
	at io.questdb.network.AbstractIODispatcher.processIOQueue(AbstractIODispatcher.java:189)
	at io.questdb.cutlass.http.HttpServer$1.run(HttpServer.java:101)
	at io.questdb.mp.Worker.run(Worker.java:118)

```

### To reproduce

_No response_

### Expected Behavior

_No response_

### Environment

```markdown
- **QuestDB version**:7.1.1
```

### Additional context

_No response_

### LINKED ISSUE #3579: Null handling in nullif(string, string)
### Describe the bug

PostgreSQL 14:
```sql
select nullif(5, null); -- returns 5. as expected
select nullif('foo', null); -- returns foo. as expected.
```
QuestDB master
```sql
select nullif(5, null) from long_sequence(1); --returns 5. the same behaviour as Postgres. :ok_hand:
select nullif('foo', null) from long_sequence(1); --returns null - differs from Postgres. :bug:
```

QuestDB should behave the same as PostgreSQL.

### PR DESCRIPTION
**Pull request by ideoma**
fixes #3563
fixes #3579

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/CharFunction.java|1|1|100.00%|

**Review by bziobrowski — DISMISSED** [review]

File: core/src/main/java/io/questdb/griffin/engine/functions/CharFunction.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ public short getShort(Record rec) {
     @Override
     public final CharSequence getStr(Record rec) {
         final char value = getChar(rec);
-        if (value == 0) {
+        if (value == Character.MIN_VALUE) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/CharFunction.java:144]: don't we use 0 as NULL for character ?

Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/engine/functions/CharFunction.java:144]: can we make sure there is correctly named constant value for it, such as CHAR_NULL or something similar?

File: core/src/main/java/io/questdb/griffin/engine/functions/cast/CastCharToSymbolFunctionFactory.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public Function newInstance(int position, ObjList<Function> args, IntList argPos
         return new Func(arg);
     }
 
-    private static class Func extends AbstractCastToSymbolFunction {
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/griffin/engine/functions/cast/CastCharToSymbolFunctionFactory.java:61]: Unrelated to this change, just something I noticed when reviewing it and maybe worth changing - See the lines 91 and 92: They create 2 different String instances out of the same `StringSink`.
```java
final String str = Chars.toString(sink);
symbols.add(Chars.toString(sink));
return str;
```


That's a waste, just a single instance would be enough.  I can push a change to this branch if that's ok with you.

**Review by jerrinot — APPROVED** [review]

**Comment by jerrinot** [comment]
a silly question: why don't we just change `CharFunction` from  
```java
@Override
public final CharSequence getSymbol(Record rec) {
    throw new UnsupportedOperationException();
}
```
to
```java
@Override
public final CharSequence getSymbol(Record rec) {
    return getStr(rec);
}
```
That's what e.g. the `StrFunction` does. 

We do not have to go through the full casting-to-symbols-whoops. I understand most of the casting-to-symbol complexity is in building a dictionary. but we don't need it, do we? We just need the string value. The test this PR is adding is also passing against `master` with the change proposed above. Is there a scenario where this would fail?

File: core/src/main/java/io/questdb/griffin/engine/functions/CharFunction.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/engine/functions/CharFunction.java]: I reverted this change, not needed

**Review by jerrinot — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 14 / 15 (93.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/Chars.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/AbstractCastToSymbolFunction.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastCharToSymbolFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/conditional/NullIfStrFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/FunctionParser.java|2|2|100.00%|

**Pull request merged**